backend/src/main/java/com/example/dontwastetomuch/user/MyUserController.java

```diff
@@ -20,7 +20,7 @@ public ResponseEntity<Void> createUser(@RequestBody MyUserCreationData userCreat
         try {
             userService.createNewUser(userCreationData);
             return ResponseEntity.status(HttpStatus.CREATED).build();
-        } catch (IllegalArgumentException e) {
+        } catch (Exception e) {
             return ResponseEntity.badRequest().build();
         }
     }
```

Comment:
last review i forgot to mention that u can have more then one Catch block

So in your case since you declare the IllegalArgumentException yourself in yout MyUserService, u could catch this extra.

```Java
try {
  //try smth
} catch (IllegalArgumentException e) {
  //catch IllegalArgumentException
} catch (Exception e) {
  //catch Other Exceptions
}

```